gax/src/main/java/com/google/api/gax/core/PropertiesProvider.java

```diff
@@ -1,119 +0,0 @@
-/*
- * Copyright 2017, Google LLC All rights reserved.
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are
- * met:
- *
- *     * Redistributions of source code must retain the above copyright
- * notice, this list of conditions and the following disclaimer.
- *     * Redistributions in binary form must reproduce the above
- * copyright notice, this list of conditions and the following disclaimer
- * in the documentation and/or other materials provided with the
- * distribution.
- *     * Neither the name of Google LLC nor the names of its
- * contributors may be used to endorse or promote products derived from
- * this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
- * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
- * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
- * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
- * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
- * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
- * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
- * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
- * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
- * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-package com.google.api.gax.core;
-
-import com.google.api.core.BetaApi;
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.Properties;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
-/** Provides meta-data properties stored in a properties file. */
-@BetaApi
-public class PropertiesProvider {
```

Comment:
I have this vague memory of going down this path because there were issues with certain contexts not having the implementation version available... Are there any contexts where the new manifest approach will break down but where the properties approach would not?

Reply:
Implementation version is not available when we run from sources. But same is true for `ProperitesProvider`, since the properties file was constructed as part of the build. 

Also, I believe reporting empty version is appropriate in case if the code is executed directly from sources - this (empty or some constant like "DEVELOPMENT") is the only true answer to the question "which version it is", because the library gets real version only when it is packaged in some sort of artifact as part of releasing process, which assigns the version.

Reply:
Here is a higher-level requirement I would like to introduce: There should be a test *somewhere* that verifies that we will get a version of the right pattern when people use our libraries. Since we won't be validating such a thing in api-client-staging, we probably need to do this validation inside google-cloud-java.